core/geometry/src/curve/internalContexts/EllipticalArcApproximationContext.ts

```diff
@@ -0,0 +1,1106 @@
+  private static workPt0 = Point3d.createZero();
+  private static workPt1 = Point3d.createZero();
+  private static workPt2 = Point3d.createZero();
+  private static workRay = Ray3d.createZero();
+
+  /** captures input */
+  private constructor(arc: Arc3d) {
+    this._isValidArc = false;
+    const data = arc.toScaledMatrix3d();
+    this._arc = Arc3d.createScaledXYColumns(data.center, data.axes, data.r0, data.r90, data.sweep);
+    this._localToWorld = Transform.createRefs(data.center, data.axes);
+    if (this._localToWorld.matrix.isSingular())
+      return;
+    if (this._arc.sweep.isEmpty)
+      return; // ellipse must have a nonzero sweep
+    const xMag2 = arc.matrixRef.columnXMagnitudeSquared();
+    const yMag2 = arc.matrixRef.columnYMagnitudeSquared();
+    if (Geometry.isSmallMetricDistanceSquared(xMag2) || Geometry.isSmallMetricDistanceSquared(yMag2))
+      return; // ellipse must have positive radii
+    if (Geometry.isSameCoordinateSquared(xMag2, yMag2))
+      return; // ellipse must not be circular
+    this._isValidArc = true;
+  }
+  /** Constructor, clones input. */
+  public static create(arc: Arc3d) {
+    return new EllipticalArcApproximationContext(arc);
+  }
+  /** The arc to be sampled. Its axes are forced to be perpendicular. */
+  public get arc(): Arc3d {
+    return this._arc;
+  }
+  /** The rigid transformation that rotates and translates the arc into the xy-plane at the origin. */
+  public get localToWorld(): Transform {
+    return this._localToWorld;
+  }
+  /** Whether the arc is amenable to sampling. */
+  public get isValidArc(): boolean {
+    return this._isValidArc;
+  }
+  /**
+   * Create a clone of the valid arc in local coordinates.
+   * @param fullSweep Optionally set full sweep on the returned local arc.
+   */
+  public cloneLocalArc(fullSweep?: boolean): Arc3d {
+    const worldToLocal = this.localToWorld.inverse()!;
+    const arcXY = this.arc.cloneTransformed(worldToLocal);
+    if (fullSweep)
+      arcXY.sweep.setStartEndRadians();
+    return arcXY;
+  }
+  /** Process structured sample data for the given elliptical arc. */
+  public static processQuadrantFractions(
+    ellipticalArc: Arc3d, samples: QuadrantFractions[], processor: QuadrantFractionsProcessor,
+  ): void {
+    const pt0 = this.workPt0;
+    const pt1 = this.workPt1;
+    const pt2 = this.workPt2;
+    const ray = this.workRay;
+    const arcBetween2Samples = (arcStart: Ray3d, arcEnd: Point3d, reverse: boolean): Arc3d => {
+      // assume non-colinear inputs
+      const myArc = Arc3d.createCircularStartTangentEnd(arcStart.origin, arcStart.direction, arcEnd) as Arc3d;
+      if (reverse)
+        myArc.reverseInPlace();
+      return myArc;
+    };
+    const createFirstArc = (f0: number, f1: number, reverse: boolean) => {
+      ellipticalArc.fractionToPointAndDerivative(f0, ray);
+      pt0.setFrom(ray.origin);
+      pt2.setFrom(ellipticalArc.fractionToPoint(f1, pt1)); // set pt2 in case no inner arcs
```

Comment:
is `pt1 = pt2`? If so, why do you even need `pt2`?

Reply:
Then I would remove 
``` 
pt2.setFrom(ellipticalArc.fractionToPoint(f1, pt1)); // set pt2 in case no inner arcs
```
and change last line to
```
return { p0: pt0, p1: pt1, p2: pt1 }; // set p2 in case no inner arcs
```

Reply:
Cannot remove that line.  It sets pt1.  And that line contains the comment.

Reply:
ok